Apps/Common/src/Services/EmailQueueService.cs

```diff
@@ -71,8 +70,7 @@ public async Task QueueNewEmailAsync(string toEmail, string templateName, bool s
         /// <inheritdoc/>
         public async Task QueueNewEmailAsync(string toEmail, string templateName, Dictionary<string, string> keyValues, bool shouldCommit = true, CancellationToken ct = default)
         {
-            EmailTemplate emailTemplate = await this.GetEmailTemplateAsync(templateName, ct) ?? throw new ProblemDetailsException(
-                ExceptionUtility.CreateServerError($"{ServiceType.Database}:{ErrorType.CommunicationInternal}", ErrorMessages.EmailTemplateNotFound));
+            EmailTemplate emailTemplate = await this.GetEmailTemplateAsync(templateName, ct) ?? throw new NotFoundException(ErrorMessages.EmailTemplateNotFound);
```

Comment:
Catch and change per calling context.

Reply:
I've deemed it simpler and more in line with how the service is truly used to go with a `DatabaseException`.

We aren't providing emailing services to an external consumer, thus this "public" method will also form a part of some internal process, with this in mind I would rather argue it will always be an internal error from the perspective of the user.